Reduce allocations / clean up code in Process (+ fix name clash with Formatter)

mathias-lang-sociomantic opened the pull request:
Importing both `Formatter` and `Process` led to a needless clash. This fixes it along with a few other code oddities.

mathias-lang-sociomantic committed:
Use the Formatter in Process

The name format defined in Process, even if private, clashed with the Formatter's format method,
so removing it should provide for a better user experience while also reducing the number of allocations
and cleaning up the code.

mathias-lang-sociomantic committed:
Make Process' Result non-allocating when possible

leandro-lucarella-sociomantic commented on src/ocean/sys/Process.d:
```diff
@@ -866,7 +856,7 @@ class Process
             // validate the redirection flags
             if((_redirect & (Redirect.OutputToError | Redirect.ErrorToOutput)) == (Redirect.OutputToError | Redirect.ErrorToOutput))
-                throw new ProcessCreateException(_args[0], "Illegal redirection flags", __FILE__, __LINE__);
+                throw new ProcessCreateException(_args[0], "Illegal redirection flags");
```
Is this really related to the use of `Formatter`?

leandro-lucarella-sociomantic commented on src/ocean/sys/Process.d:
```diff
@@ -1596,14 +1586,12 @@ class Process
 {
-    public this(cstring command, istring file, uint line)
+    public this (cstring command, istring message = SysError.lastMsg,
+                 istring file = __FILE__, uint line = __LINE__)
```
I guess not, can you split this change in a separate commit?

mathias-lang-sociomantic committed:
Improve Exceptions in ocean.sys.Process

The ctors had two issues: the main one, they were not forwarding their file and line arguments, so it was lost.
The second one, they did not provide default __FILE__ and __LINE__ default arguments as is common with exceptions ctor,
leading to rather non-DRY code.
The two ctors of ProcessCreateException have been merged, as a call to new ProcessCreateException(istring, istring) was ambiguous
after the addition of default parameters.

mathias-lang-sociomantic commented:
Splitted changes to be more atomic

mihails-strasuns-sociomantic reviewed (approved):
Thanks, very nice!
